nasapp/source/cups-2.0.0/man/lprm.man
.\"
.\" "$Id: lprm.man,v 1.1.1.1 2014/12/23 05:45:37 thki81 Exp $"
.\"
.\" lprm man page for CUPS.
.\"
.\" Copyright 2007-2014 by Apple Inc.
.\" Copyright 1997-2006 by Easy Software Products.
.\"
.\" These coded instructions, statements, and computer programs are the
.\" property of Apple Inc. and are protected by Federal copyright
.\" law.  Distribution and use rights are outlined in the file "LICENSE.txt"
.\" which should have been included with this file.  If this file is
.\" file is missing or damaged, see the license at "http://www.cups.org/".
.\"
.TH lprm 1 "CUPS" "22 May 2014" "Apple Inc."
.SH NAME
lprm \- cancel print jobs
.SH SYNOPSIS
.B lprm
[
.B \-E
] [
.B \-U
.I username
] [
.B \-h
.IR server [ :port ]
] [
.B \-P
.IR destination [ /instance ]
] [
.B \-
] [
.I job-id(s)
]
.SH DESCRIPTION
.B lprm
cancels print jobs that have been queued for printing.
If no arguments are supplied, the current job on the default destination is canceled.
You can specify one or more job ID numbers to cancel those jobs or use the \fI\-\fR option to cancel all jobs.
.SH OPTIONS
The
.B lprm
command supports the following options:
.TP 5
.B \-E
Forces encryption when connecting to the server.
.TP 5
\fB\-P \fIdestination\fR[\fI/instance\fR]
Specifies the destination printer or class.
.TP 5
\fB\-U \fIusername\fR
Specifies an alternate username.
.TP 5
\fB\-h \fIserver\fR[\fI:port\fR]
Specifies an alternate server.
.SH CONFORMING TO
The CUPS version of
.B lprm
is compatible with the standard Berkeley command of the same name.
.SH EXAMPLES
Cancel the current job on the default printer:
.nf

    lprm

.fi
Cancel job 1234:
.nf

    lprm 1234

.fi
Cancel all jobs:
.nf

    lprm \-
.fi
.SH SEE ALSO
.BR cancel (1),
.BR lp (1),
.BR lpq (1),
.BR lpr (1),
.BR lpstat (1),
CUPS Online Help (http://localhost:631/help)
.SH COPYRIGHT
Copyright \[co] 2007-2014 by Apple Inc.
.\"
.\" End of "$Id: lprm.man,v 1.1.1.1 2014/12/23 05:45:37 thki81 Exp $".
.\"


poptop-1.1.4/ChangeLog
Poptop ChangeLog
---------------------------------------------------------------------------
v1.1.4
* Wed Apr  9 2003 Richard de Vroede <r.devroede@linvision.com>
- fixed a potential buffer-overflow in ctrlpacket.c
- callid enhancements
* Fri Oct 11 2002 Richard de Vroede <r.devroede@linvision.com>
- added BCrelay (Broadcast relay)
---------------------------------------------------------------------------
v1.1.3
* Thu Aug 22 2002 Richard de Vroede <r.devroede@linvision.com>
- added stimeout option to pptpd.conf manpage
- updated the Changelog file ;-)

* Tue Aug 20 2002 Richard de Vroede <r.devroede@linvision.com>
- removed debug commandline option from pptpd.init

* Thu Aug  1 2002 Richard de Vroede <r.devroede@linvision.com>
- added config(noreplace) so old configs don't get replaced
- fixed postscriptlet
- adapted spec to cvs tree

* Wed Jun 26 2002 Richard de Vroede <r.devroede@linvision.com>
- specfile now supports --with[out] options

---------------------------------------------------------------------------


nasapp/source/cups-2.0.0/berkeley/lprm.c
/*
 * "$Id: lprm.c,v 1.1.1.1 2014/12/23 05:45:30 thki81 Exp $"
 *
 *   "lprm" command for CUPS.
 *
 *   Copyright 2007-2010 by Apple Inc.
 *   Copyright 1997-2006 by Easy Software Products.
 *
 *   These coded instructions, statements, and computer programs are the
 *   property of Apple Inc. and are protected by Federal copyright
 *   law.  Distribution and use rights are outlined in the file "LICENSE.txt"
 *   which should have been included with this file.  If this file is
 *   file is missing or damaged, see the license at "http://www.cups.org/".
 *
 * Contents:
 *
 *   main() - Parse options and cancel jobs.
 */

/*
 * Include necessary headers...
 */

#include <cups/cups-private.h>


/*
 * 'main()' - Parse options and cancel jobs.
 */

int				/* O - Exit status */
main(int  argc,			/* I - Number of command-line arguments */
     char *argv[])		/* I - Command-line arguments */
{
  int		i;		/* Looping var */
  int		job_id;		/* Job ID */
  const char	*name;		/* Destination printer */
  char		*instance;	/* Pointer to instance name */
  cups_dest_t	*dest,		/* Destination */
		*defdest;	/* Default destination */
  int		did_cancel;	/* Did we cancel something? */


  _cupsSetLocale(argv);

 /*
  * Setup to cancel individual print jobs...
  */

  did_cancel = 0;
  defdest    = cupsGetNamedDest(CUPS_HTTP_DEFAULT, NULL, NULL);
  name       = defdest ? defdest->name : NULL;

 /*
  * Process command-line arguments...
  */

  for (i = 1; i < argc; i ++)
    if (argv[i][0] == '-' && argv[i][1] != '\0')
      switch (argv[i][1])
      {
        case 'E' : /* Encrypt */
#ifdef HAVE_SSL
	    cupsSetEncryption(HTTP_ENCRYPT_REQUIRED);
#else
            _cupsLangPrintf(stderr,
	                    _("%s: Sorry, no encryption support."), argv[0]);
#endif /* HAVE_SSL */
	    break;

        case 'P' : /* Cancel jobs on a printer */
	    if (argv[i][2])
	      name = argv[i] + 2;
	    else
	    {
	      i ++;
	      name = argv[i];
	    }

	    if ((instance = strchr(name, '/')) != NULL)
	      *instance = '\0';

	    if ((dest = cupsGetNamedDest(CUPS_HTTP_DEFAULT, name,
	                                 NULL)) == NULL)
	    {
	      _cupsLangPrintf(stderr,
	                      _("%s: Error - unknown destination \"%s\"."),
			      argv[0], name);
              goto error;
	    }

	    cupsFreeDests(1, dest);
	    break;

        case 'U' : /* Username */
	    if (argv[i][2] != '\0')
	      cupsSetUser(argv[i] + 2);
	    else
	    {
	      i ++;
	      if (i >= argc)
	      {
	        _cupsLangPrintf(stderr,
		                _("%s: Error - expected username after "
				  "\"-U\" option."), argv[0]);
	        goto error;
	      }

              cupsSetUser(argv[i]);
	    }
	    break;
	    
        case 'h' : /* Connect to host */
	    if (argv[i][2] != '\0')
              cupsSetServer(argv[i] + 2);
	    else
	    {
	      i ++;

	      if (i >= argc)
	      {
	        _cupsLangPrintf(stderr,
		        	_("%s: Error - expected hostname after "
			          "\"-h\" option."), argv[0]);
		goto error;
              }
	      else
                cupsSetServer(argv[i]);
	    }

            if (defdest)
	      cupsFreeDests(1, defdest);

	    defdest = cupsGetNamedDest(CUPS_HTTP_DEFAULT, NULL, NULL);
	    name    = defdest ? defdest->name : NULL;
	    break;

	default :
	    _cupsLangPrintf(stderr, _("%s: Error - unknown option \"%c\"."),
			    argv[0], argv[i][1]);
            goto error;
      }
    else
    {
     /*
      * Cancel a job or printer...
      */

      if ((dest = cupsGetNamedDest(CUPS_HTTP_DEFAULT, argv[i], NULL)) != NULL)
        cupsFreeDests(1, dest);

      if (dest)
      {
        name   = argv[i];
        job_id = 0;
      }
      else if (isdigit(argv[i][0] & 255))
      {
        name   = NULL;
        job_id = atoi(argv[i]);
      }
      else if (!strcmp(argv[i], "-"))
      {
       /*
        * Cancel all jobs
        */

        job_id = -1;
      }
      else
      {
	_cupsLangPrintf(stderr, _("%s: Error - unknown destination \"%s\"."),
			argv[0], argv[i]);
	goto error;
      }

      if (cupsCancelJob2(CUPS_HTTP_DEFAULT, name, job_id, 0) != IPP_OK)
      {
        _cupsLangPrintf(stderr, "%s: %s", argv[0], cupsLastErrorString());
	goto error;
      }

      did_cancel = 1;
    }

 /*
  * If nothing has been canceled yet, cancel the current job on the specified
  * (or default) printer...
  */

  if (!did_cancel && cupsCancelJob2(CUPS_HTTP_DEFAULT, name, 0, 0) != IPP_OK)
    {
      _cupsLangPrintf(stderr, "%s: %s", argv[0], cupsLastErrorString());
      goto error;
    }

  if (defdest)
    cupsFreeDests(1, defdest);

  return (0);

 /*
  * If we get here there was an error, so clean up...
  */

  error:

  if (defdest)
    cupsFreeDests(1, defdest);

  return (1);
}


/*
 * End of "$Id: lprm.c,v 1.1.1.1 2014/12/23 05:45:30 thki81 Exp $".
 */
